libs/datamodel/core/tests/attributes/default_composite_negative.rs

```diff
@@ -0,0 +1,411 @@
+          [1;94m-->[0m  [4mschema.prisma:2[0m
+        [1;94m   | [0m
+        [1;94m 1 | [0mtype Composite {
+        [1;94m 2 | [0m  enum A @[1;91mdefault(B)[0m
+        [1;94m   | [0m
+    "#]];
+
+    expectation.assert_eq(&error)
+}
+
+#[test]
+fn must_error_if_scalar_default_on_unsupported() {
+    let dml = indoc! {r#"
+        datasource db1 {
+          provider = "postgresql"
+          url = "postgresql://"
+        }
+
+        type Composite {
+          balance Unsupported("some random stuff") @default(12)
+        }
+    "#};
+
+    let error = datamodel::parse_schema(dml).map(drop).unwrap_err();
+
+    let expectation = expect![[r#"
+        [1;91merror[0m: [1mError parsing attribute "@default": Composite field of type `Unsupported` cannot have default values.[0m
+          [1;94m-->[0m  [4mschema.prisma:7[0m
+        [1;94m   | [0m
+        [1;94m 6 | [0mtype Composite {
+        [1;94m 7 | [0m  balance Unsupported("some random stuff") @[1;91mdefault(12)[0m
+        [1;94m   | [0m
+    "#]];
+
+    expectation.assert_eq(&error)
+}
+
+#[test]
+fn named_default_values_are_not_allowed() {
+    let dml = indoc! { r#"
+        type A {
+          id String @default(cuid(), map: "nope__nope__nope")
+        }
+    "#};
+
+    let error = datamodel::parse_schema(dml).map(drop).unwrap_err();
+
+    let expectation = expect![[r#"
+        [1;91merror[0m: [1mError parsing attribute "@default": The `map` argument is not allowed on a composite type field.[0m
+          [1;94m-->[0m  [4mschema.prisma:2[0m
+        [1;94m   | [0m
+        [1;94m 1 | [0mtype A {
+        [1;94m 2 | [0m  id String @[1;91mdefault(cuid(), map: "nope__nope__nope")[0m
+        [1;94m   | [0m
+    "#]];
+
+    expectation.assert_eq(&error)
+}
+
+#[test]
+fn default_on_composite_type_field_errors() {
+    let schema = indoc! { r#"
+        datasource db {
+            provider = "mongodb"
+            url = "mongodb://"
+        }
+
+        generator client {
+            provider = "prisma-client-js"
+            previewFeatures = ["mongoDb"]
+        }
+
+        type Address {
+            street String
+        }
+
+        type User {
+            address Address? @default("{ \"street\": \"broadway\"}")
+        }
+    "#};
+
+    let error = datamodel::parse_schema(schema).map(drop).unwrap_err();
+
+    let expected = expect![[r#"
+        [1;91merror[0m: [1mError validating field `address` in composite type `Address`: Defaults on fields of type composite are not supported. Please remove the `@default` attribute.[0m
+          [1;94m-->[0m  [4mschema.prisma:16[0m
+        [1;94m   | [0m
+        [1;94m15 | [0mtype User {
+        [1;94m16 | [0m    address Address? @[1;91mdefault("{ \"street\": \"broadway\"}")[0m
+        [1;94m   | [0m
+    "#]];
+
+    expected.assert_eq(&error)
+}
+
+#[test]
+fn must_error_on_dbgenerated_default_on_non_native_type_on_mongodb() {
```

Comment:
we can be a lot more general here: `@default(dbgenerated(...))` is not supported on composite type fields.